internal/pkg/template/input_test.go

```diff
@@ -0,0 +1,236 @@
+		Type:        "string",
+		Default:     "def",
+		Options:     []Option{"a", 1},
+		Kind:        "input",
+	}}
+	err = inputs.Validate()
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), `key="options"`)
+
+	// Success - with Options
+	inputs = Inputs{{
+		Id:          "fb.extractor.password",
+		Name:        "input",
+		Description: "input desc",
+		Type:        "string",
+		Default:     "def",
+		Options:     []Option{"a", "b"},
+		Kind:        "select",
+	}}
+	err = inputs.Validate()
+	assert.NoError(t, err)
+
+	// Success - int Default and empty Options
+	inputs = Inputs{{
+		Id:          "fb.extractor.password",
+		Name:        "input",
+		Description: "input desc",
+		Type:        "int",
+		Default:     33,
+		Options:     []Option{},
+		Kind:        "input",
+	}}
+	err = inputs.Validate()
+	assert.NoError(t, err)
+
+	// Success - no Default
+	inputs = Inputs{{
+		Id:          "fb.extractor.password",
+		Name:        "input",
+		Description: "input desc",
+		Type:        "string",
+		Kind:        "input",
+	}}
+	err = inputs.Validate()
+	assert.NoError(t, err)
+}
+
+func TestTemplateInputsJsonUnmarshal(t *testing.T) {
+	t.Parallel()
+
+	inputsJson := `
+[
+  {
+    "id": "fb.extractor.username",
+    "name": "Facebook username",
+    "description": "Facebook username description",
+    "type": "string",
+    "kind": "input"
+  },
+  {
+    "id": "fb.extractor.password",
+    "name": "Facebook password",
+    "description": "Facebook password description",
+    "type": "string",
+    "options": [],
+    "kind": "password"
+  },
+  {
+    "id": "fb.extractor.option",
+    "name": "Facebook option",
+    "description": "Facebook option description",
+    "type": "int",
+    "options": [1, 2],
+    "kind": "select"
+  }
+]`
+	var inputs Inputs
+	assert.NoError(t, json.Unmarshal([]byte(inputsJson), &inputs))
+	assert.Len(t, inputs, 3)
+
+	assert.IsType(t, &Input{}, inputs[0])
+	assert.Equal(t, inputs[0].Id, "fb.extractor.username")
+	assert.Equal(t, inputs[0].Name, "Facebook username")
+	assert.Equal(t, inputs[0].Description, "Facebook username description")
+	assert.Equal(t, inputs[0].Type, "string")
+	assert.Equal(t, inputs[0].Kind, "input")
+	assert.Len(t, inputs[1].Options, 0)
+
+	assert.IsType(t, &Input{}, inputs[1])
+	assert.Equal(t, inputs[1].Id, "fb.extractor.password")
+	assert.Equal(t, inputs[1].Name, "Facebook password")
+	assert.Equal(t, inputs[1].Description, "Facebook password description")
+	assert.Equal(t, inputs[1].Type, "string")
+	assert.Equal(t, inputs[1].Kind, "password")
+	assert.IsType(t, inputs[1].Options, []Option{})
+	assert.Len(t, inputs[1].Options, 0)
+
+	assert.IsType(t, &Input{}, inputs[2])
+	assert.Equal(t, inputs[2].Id, "fb.extractor.option")
+	assert.Equal(t, inputs[2].Name, "Facebook option")
+	assert.Equal(t, inputs[2].Description, "Facebook option description")
+	assert.Equal(t, inputs[2].Type, "int")
+	assert.Equal(t, inputs[2].Kind, "select")
+	assert.IsType(t, inputs[2].Options, []Option{1, 2})
```

Comment:
Toto by bolo prehladnejsie assert tak, ze si spravis nejaku `expected` variable a tam bude:
```
expected := Inputs{
  {
    Id: "....",
  },
}
```

A potom `assert.Equal`.

Go vie pekne vypisat v teste diff, kde je rozdiel. A ak by si na nejaky kluc zabudol, ... tak bude mat urcite default hodnotu -> ak test prejde.

Pritom teraz ak sa na nejaky kluc zabudne, tak moze mat lubovolnu hodnotu.

Reply:
Vlastne tak, ako to mas v teste `TestTemplateInputsJsonMarshal(` len naopak.

Reply:
ok 👍